provisioner/ansible-local/provisioner.go

```diff
@@ -23,51 +24,126 @@ const DefaultStagingDir = "/tmp/packer-provisioner-ansible-local"
 type Config struct {
 	common.PackerConfig `mapstructure:",squash"`
 	ctx                 interpolate.Context
-
-	// The command to run ansible
-	Command string
-
-	// Extra options to pass to the ansible command
+	// The command to invoke ansible. Defaults to
+	//  `ansible-playbook`. If you would like to provide a more complex command,
+	//  for example, something that sets up a virtual environment before calling
+	//  ansible, take a look at the ansible wrapper guide below for inspiration.
+	//  Please note that Packer expects Command to be a path to an executable.
+	//  Arbitrary bash scripting will not work and needs to go inside an
+	//  executable script.
+	Command string `mapstructure:"command"`
+	// Extra arguments to pass to Ansible.
+	// These arguments _will not_ be passed through a shell and arguments should
+	// not be quoted. Usage example:
+	//
+	// ```json
+	//   "extra_arguments": [ "--extra-vars", "Region={{user `Region`}} Stage={{user `Stage`}}" ]
+	// ```
+	// In certain scenarios where you want to pass ansible command line arguments
+	// that include parameter and value (for example `--vault-password-file pwfile`),
+	// from ansible documentation this is correct format but that is NOT accepted here.
+	// Instead you need to do it like `--vault-password-file=pwfile`.
+	//
+	// If you are running a Windows build on AWS, Azure, Google Compute, or OpenStack
+	// and would like to access the auto-generated password that Packer uses to
+	// connect to a Windows instance via WinRM, you can use the template variable
+	// `{{.WinRMPassword}}` in this option. For example:
+	//
+	// ```json
+	//   "extra_arguments": [
+	//     "--extra-vars", "winrm_password={{ .WinRMPassword }}"
+	//   ]
+	// ```
 	ExtraArguments []string `mapstructure:"extra_arguments"`
-
-	// Path to group_vars directory
+	// A path to the directory containing ansible group
+    // variables on your local system to be copied to the remote machine. By
+    // default, this is empty.
 	GroupVars string `mapstructure:"group_vars"`
-
-	// Path to host_vars directory
+	// A path to the directory containing ansible host variables on your local
+	// system to be copied to the remote machine. By default, this is empty.
 	HostVars string `mapstructure:"host_vars"`
-
-	// The playbook dir to upload.
+	// A path to the complete ansible directory structure on your local system
+	// to be copied to the remote machine as the `staging_directory` before all
+	// other files and directories.
 	PlaybookDir string `mapstructure:"playbook_dir"`
-
-	// The main playbook file to execute.
+	// The playbook file to be executed by ansible. This file must exist on your
+	// local system and will be uploaded to the remote machine. This option is
+	// exclusive with `playbook_files`.
 	PlaybookFile string `mapstructure:"playbook_file"`
```

Comment:
If you add `required:"true"` it will generate a specific mdx partial with the required config. 
```suggestion
	// The playbook file to be executed by ansible. This file must exist on your
	// local system and will be uploaded to the remote machine. This option is
	// exclusive with `playbook_files`.
	PlaybookFile string `mapstructure:"playbook_file" required:"true"`
```

Reply:
I understood that. But it's not required if you specify `playbook_files` (see the original docs)

Reply:
Oh, you're right! it's required only if the other is not set. I got confused 👀